alf/utils/summary_utils.py

```diff
@@ -108,47 +78,20 @@ def histogram_continuous(name,
         bucket_max (float|None): represent bucket max value,
             if None value tf.reduce_max(data) will be used
         bucket_count (int):  positive `int`. The output will have this many buckets.
-        step (None|tf.Variable):  step value for this summary. this defaults to
-            `tf.summary.experimental.get_step()`
-        description (str): Optional long-form description for this summary
+        step (None|Tensor): step value for this summary. this defaults to
+            `alf.summary.get_global_counter()`
     """
-    # TODO: implement pytorch version
-    return
-    summary_metadata = metadata.create_summary_metadata(
-        display_name=None, description=description)
-    summary_scope = (getattr(tf.summary.experimental, 'summary_scope', None)
-                     or tf.summary.summary_scope)
-    with summary_scope(
-            name,
-            'histogram_summary',
-            values=[data, bucket_min, bucket_max, bucket_count, step]) as (tag,
-                                                                           _):
-        with tf.name_scope('buckets'):
-            data = tf.cast(tf.reshape(data, shape=[-1]), tf.float64)
-            if bucket_min is None:
-                bucket_min = tf.reduce_min(data)
-            if bucket_max is None:
-                bucket_max = tf.reduce_min(data)
-            range_ = bucket_max - bucket_min
-            bucket_width = range_ / tf.cast(bucket_count, tf.float64)
-            offsets = data - bucket_min
-            bucket_indices = tf.cast(
-                tf.floor(offsets / bucket_width), dtype=tf.int32)
-            clamped_indices = tf.clip_by_value(bucket_indices, 0,
-                                               bucket_count - 1)
-            one_hots = tf.one_hot(clamped_indices, depth=bucket_count)
-            bucket_counts = tf.cast(
-                tf.reduce_sum(input_tensor=one_hots, axis=0), dtype=tf.float64)
-            edges = tf.linspace(bucket_min, bucket_max, bucket_count + 1)
-            edges = tf.concat([edges[:-1], [tf.cast(bucket_max, tf.float64)]],
-                              0)
-            edges = tf.cast(edges, tf.float64)
-            left_edges = edges[:-1]
-            right_edges = edges[1:]
-            tensor = tf.transpose(
-                a=tf.stack([left_edges, right_edges, bucket_counts]))
-        return tf.summary.write(
-            tag=tag, tensor=tensor, step=step, metadata=summary_metadata)
+    data = data.to(torch.float64)
+    if bucket_min is None:
+        bucket_min = data.min()
+    if bucket_max is None:
+        bucket_max = data.max()
+    bins = (
+        bucket_min +
+        (torch.arange(bucket_count + 1, dtype=torch.float64) / bucket_count) *
+        (bucket_max - bucket_min))
+    data = data.clamp(bucket_min, bucket_max)
+    alf.summary.histogram(name, data, step=step, bins=bins.cpu())
```

Comment:
not sure why .cpu() is needed.

Reply:
it complains not cpu somewhere.

Reply:
Could it be some other place is returning cpu instead of gpu tensors, like the environment case?
Ok to merge this change, just want to make sure there are no underlying problems.

Reply:
bins indeed can be cuda.  The failure is because pytorch's tensorboard API want to call numpy() (fail if not cpu) on the tensor, which can be a cuda tensor.